Sources/TensorFlow/Core/StringTensor.swift

```diff
@@ -46,50 +46,72 @@ extension StringTensor {
 
     // utf8CString is null-terminated. TF APIs want the strings without null-terminators.
     let cStrings = scalars.map { $0.utf8CString.dropLast() }
-    let tfEncodedSizes = cStrings.map { TF_StringEncodedSize($0.count) }
-
-    // Format information copied from tensorflow/c/c_api.h:
-    // The format for TF_STRING tensors is:
-    //   start_offset: array[uint64]
-    //   data:         byte[...]
-    //
-    //   The string length (as a varint), followed by the contents of the string is encoded at
-    //   data[start_offset[i]]].
-    // The size of the "start_offset" region.
-    let startOffsetsByteCount = scalars.count * MemoryLayout<UInt64>.stride
-
-    // The size of the "data" region.
-    let dataByteCount = tfEncodedSizes.reduce(0, +) * MemoryLayout<UInt8>.stride
-
-    let handle = TensorHandle<String>(
-      shape: shape.dimensions,
-      byteCount: startOffsetsByteCount + dataByteCount,
-      bufferInitializer: { tensorBuffer in
-        // Initialize the "start_offset" region.
-        var startOffset: UInt64 = 0
-        var startOffsetAddr =
-          tensorBuffer.bindMemory(to: UInt64.self, capacity: scalars.count)
-        for tfEncodedSize in tfEncodedSizes {
-          startOffsetAddr.initialize(to: startOffset)
-          startOffsetAddr = startOffsetAddr.advanced(by: 1)
-          startOffset = startOffset + UInt64(tfEncodedSize)
-        }
-
-        // Initialize the "data" region.
-        var dataAddr = tensorBuffer.advanced(by: startOffsetsByteCount)
-          .bindMemory(to: Int8.self, capacity: dataByteCount)
-        let status = TF_NewStatus()
-        for (cString, tfEncodedSize) in zip(cStrings, tfEncodedSizes) {
-          _ = cString.withUnsafeBufferPointer { buffer in
-            TF_StringEncode(
-              buffer.baseAddress, buffer.count, dataAddr, tfEncodedSize, status)
+
+    // Note: `TENSORFLOW_MASTER` changes below are necessary for the new TensorFlow ABI-stable
+    // unified string tensor design.
+    #if TENSORFLOW_MASTER
```

Comment:
Note: Swift for TensorFlow is currently pinned to TensorFlow tag [r2.2](https://github.com/tensorflow/swift-apis/blob/b71c024ff1935e0552b3e44107c647fb6ed24c9c/CMakeLists.txt), so we'll need to add some `#if TENSORFLOW_MASTER` flags to selectively enable the new string tensor behavior until we update the TensorFlow version.